### DISCUSSION TITLE
Is there any support to trim transparencies from the borders?

### DISCUSSION THREAD

**Discussion by MarcosAndre** [libvips/libvips#3497] [Q&A]
Hello guys,

Firstly, I want to thank you for all the work done in libvips.

Is there any support to trim transparencies from the borders? 
I have multiple images with transparent borders (alpha=0.0) and I want to trim them.

Thank you in advance.
Best regards,
Marcos André

**Comment by jcupitt** [comment]
Hi @MarcosAndre,

Sure, with this test image:

![PNG_transparency_demonstration](https://github.com/libvips/libvips/assets/580843/e6c504d8-a84d-4ef4-b540-8c2d24b95d6f)

I see:

```
$ vips find_trim PNG_transparency_demonstration.png
112
22
561
504
```

Which is left, top, width, height of the image area.

This is searching for pixels with the value (0, 0, 0, 0), so it won't work if your image has junk hidden in the RGB bands. In this case it's better to just extract the alpha band and run `find_trim` on that. 

I'd use something like python, so eg.:

```
$ python
Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyvips
>>> x = pyvips.Image.new_from_file("PNG_transparency_demonstration.png")
>>> x[3].find_trim(background=0)
[112, 22, 561, 504]
>>> left, top, width, height = x[3].find_trim()
>>> top
22
>>> y = x.crop(*x[3].find_trim(background=0))
>>> y
<pyvips.Image 561x504 uchar, 4 bands, srgb>
>>> y.write_to_file("y.png")
>>> 
```

To make:

![y](https://github.com/libvips/libvips/assets/580843/b7ed0c8b-1b6f-4991-903f-2148984bb96b)

**Reply by MarcosAndre** [reply]
Thank you for your quick reply.
It works as expected. extract band did the trick. 

Cheers!
